### PR TITLE
Proof of concept: login a user via default browser

### PR DESCRIPTION
**Pull request by schacon**
This is a somewhat working example of how to use the API to log in via a default OS browser. It requests a temp login token on page load (we should probably do this just before redirecting to the login screen, but that's a little difficult because `window.open()` doesn't work in <script> in Tauri), replaces the href in a log in link, then starts polling to see if you've logged in. When it sees you have, it stops polling and shows you the response, which eventually we should store to disk and then you're *logged in*.

Currently it is hardcoded to my local box via ngrok, but we can change that to a variable or the staging server or something.

Don't actually merge this, I just wanted a place to put the more or less working code so we could discuss the approach.

### CODE REVIEW

File: src-tauri/tauri.conf.json [hunk 1]
```diff
@@ -21,6 +21,9 @@
         "all": false,
         "startDragging": true
       },
+      "shell": {
```
Comment (ngalaiko) [inline src-tauri/tauri.conf.json:24]: why did you add this permission?

File: src-tauri/src/main.rs [hunk 2]
```diff
@@ -315,6 +315,9 @@ fn main() {
                         projects_storage,
                     });
 
+                    let window = app.get_window("main").unwrap();
+                    window.open_devtools();
```
Comment (ngalaiko) [inline src-tauri/src/main.rs:319]: nice! i was thinking looking into some sort of flag to only do this in dev, but we can keep it for now

File: src/routes/user/+page.svelte [hunk 3]
```diff
@@ -0,0 +1,64 @@
+<script>
+  let response = null;
+  let interval = null;
+  let token = null;
+
+  let login;
+  let user;
```
Comment (ngalaiko) [inline src/routes/user/+page.svelte:7]: i understand this is a poc, but this should be stored on disk, right? also, the user info should be stored globally for the frontend app, so it's easy to access

File: src/routes/user/+page.svelte [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        console.log(user);
+        clearInterval(interval);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const logIn = () => {
+    postData("https://chacons.eu.ngrok.io/api/login/token")
```
Comment (ngalaiko) [inline src/routes/user/+page.svelte:37]: is this on gitbutler.com?

**Review by ngalaiko — COMMENTED** [review]
good as a poc, let me know if you need some help with a proper setup

File: src-tauri/tauri.conf.json (hunk 1 above)
Comment (schacon) [reply src-tauri/tauri.conf.json:24]: I was having problems getting it to open an external browser when you click on the link, I think I needed this to get it to work.

File: src-tauri/src/main.rs (hunk 2 above)
Comment (schacon) [reply src-tauri/src/main.rs:319]: Yeah, I needed it for debugging, a flag would be great.

File: src/routes/user/+page.svelte (hunk 3 above)
Comment (schacon) [reply src/routes/user/+page.svelte:7]: right, this should be stored globally on disk as like `user.json` or something.

File: src/routes/user/+page.svelte (hunk 4 above)
Comment (schacon) [reply src/routes/user/+page.svelte:37]: It is now, at `https://app.gitbutler.com/api`

**Pull request closed without merging**